Azure/sonic-buildimage
# Upgrade ifupdown2 to version 1.2.8

**- What I did**
Instead of using ifupdown2 package from Debian stretch repository, source compiled and installed lates
version of ifupdown2. The latest version is required to support ZTP over IPv6 transport in a stable and
quick manner.

**- How I did it**

Created a new rules/ifupdown2.mk file and compiled ifupdown2 as a SONIC_MAKE_DEBS. 
Modified slave.mk to have the ifupdown2 as a dependency to create sonic installer image.
Modified "files/build_templates/sonic_debian_extension.j2" to install source compiled ifupdown2 deb file and its dependencies.


**- How to verify it**
Load an image with these changes and check ifupdown2 version using the command dpkg -l ifupdown2

Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  ifupdown2      1.2.8-1      all          Network Interface Management tool

Verified that "systemctl restart interfaces-config" also works.
Verified if eth0 IP is picked up via DHCP without any issues.

**- Description for the changelog**

Upgrade ifupdown2 to version 1.2.8

**- A picture of a cute animal (not mandatory but encouraged)**
